src/pipe.cpp

```diff
@@ -88,6 +88,17 @@ void zmq::send_routing_id (pipe_t *pipe_, const options_t &options_)
     pipe_->flush ();
 }
 
+void zmq::send_hello_msg (pipe_t *pipe_, const options_t &options_)
+{
+    zmq::msg_t hello;
+    const int rc = hello.init_size (options_.hello_msg.size ());
+    errno_assert (rc == 0);
+    memcpy (hello.data (), &options_.hello_msg[0], options_.hello_msg.size ());
```

Comment:
memcopy to NULL with 0 size is undefined behavior and will break badly (yes, really). Probably a good place to use the new zmq::msg_t::init_buffer

Reply:
fixed